src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java

```diff
@@ -0,0 +1,270 @@
+import uk.gov.hmcts.ccd.fta.util.JsonUtils;
+
+@SuppressWarnings({"LocalVariableName"})
+public class BackEndFunctionalTestScenarioPlayer implements BackEndFunctionalTestAutomationDSL {
+
+    private static final String DYNAMIC_CONTENT_PLACEHOLDER = "[[DYNAMIC]]";
+
+    private final BackEndFunctionalTestScenarioContext scenarioContext;
+    private final AATHelper aat;
+    private Scenario scenario;
+
+    private Logger logger = LoggerFactory.getLogger(BackEndFunctionalTestScenarioPlayer.class);
+
+    public BackEndFunctionalTestScenarioPlayer() {
+        aat = AATHelper.INSTANCE;
+        RestAssured.baseURI = aat.getTestUrl();
+        RestAssured.useRelaxedHTTPSValidation();
+        scenarioContext = new BackEndFunctionalTestScenarioContext();
+    }
+
+    @Before()
+    public void prepare(Scenario scenario) {
+        this.scenario = scenario;
+    }
+
+    @Override
+    @Given("an appropriate test context as detailed in the test data source")
+    public void initializeAppropriateTestContextAsDetailedInTheTestDataSource() {
+        scenarioContext.initializeTestDataFor(scenario);
+        String logPrefix = scenarioContext.getCurrentScenarioTag() + ": Test data ";
+        if (scenarioContext.getTestData() != null) {
+            logger.info(logPrefix + "was loaded successfully");
+        } else {
+            logger.info(logPrefix + "was not found");
+        }
+    }
+
+    @Override
+    @Given("a user with [{}]")
+    public void verifyThatThereIsAUserInTheContextWithAParticularSpecification(String specificationAboutAUser) {
+        UserData aUser = scenarioContext.getTestData().getUser();
+        String username = aUser.getUsername();
+
+        // ADD MORE IDAM USERS HERE
+        if (username.equals(aat.getCaseworkerAutoTestEmail())) {
+            aUser.setPassword(aat.getCaseworkerAutoTestPassword());
+        } else {
+            logger.info(scenarioContext.getCurrentScenarioTag() + ": Idam user not recognised by FTA player");
+        }
+
+        String logPrefix = scenarioContext.getCurrentScenarioTag() + ": Idam user [" + aUser.getUsername()
+            + "][" + aUser.getPassword() + "] ";
+        try {
+            AuthenticatedUser authenticatedUserMetadata = aat.getIdamHelper().authenticate(
+                aUser.getUsername(), aUser.getPassword());
+            aUser.setToken(authenticatedUserMetadata.getAccessToken());
+            aUser.setUid(authenticatedUserMetadata.getId());
+            logger.info(logPrefix + "authenticated");
+        } catch (FeignException ex) {
+            logger.info(logPrefix + "credentials invalid");
+        }
+
+        scenarioContext.setTheUser(aUser);
+
+        boolean doesTestDataMeetSpec = scenarioContext.getTestData().meetsSpec(specificationAboutAUser);
+        String errorMessage = "Test data does not confirm it meets the specification about a user: "
+            + specificationAboutAUser;
+        Assert.assertTrue(errorMessage, doesTestDataMeetSpec);
```

Comment:
would be better to throw a (custom?) exception here instead of using an assert since we are in a prerequisite step. printing some of the TestData object in the exception could also be useful for any future debugging in case of data mismatch

Reply:
👍 created a custom exception for functional tests and throw it instead of using assertions

Reply:
yes @MSancaktutar this is what I was thinking